kitsune/questions/api.py

```diff
@@ -191,6 +195,13 @@ def filter_is_solved(self, queryset, value):
             solved_filter = ~solved_filter
         return queryset.filter(solved_filter)
 
+    def filter_solved_by(self, queryset, username):
+        question_user_solved = (
+            Question.objects.filter(solution__creator__username=username).values('id'))
+
+        solved_by_filter = Q(id__in=question_user_solved)
```

Comment:
Is this that same trick @mythmon did to force a subquery instead of a join? Or why didnt you just use the queryset above?